### PR TITLE
Add DataStore & Remember dark theme

### PR DESCRIPTION
**Pull request by JakubGongol**

### CODE REVIEW

File: app/src/main/java/com/strv/movies/MainViewModel.kt [hunk 1]
```diff
@@ -2,28 +2,28 @@ package com.strv.movies
...
-import kotlinx.coroutines.flow.asStateFlow
-import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
 
 class MainViewModel @AssistedInject constructor(
-    @Assisted savedStateHandle: SavedStateHandle,
-    @Assisted isDarkTheme: Boolean,
-): ViewModel() {
+    private val moviesDataStore: MoviesDataStore,
+    @Assisted savedStateHandle: SavedStateHandle
```
Comment (jipariz) [inline app/src/main/java/com/strv/movies/MainViewModel.kt:14]: We can now remove assisted inject. SavedStateHandle can be injected automatically by hilt when annotated with `@HiltViewModel`
Unless we would want to demonstrate assisted inject on it on my lesson, but I don't think this is the best use case for it

File: app/src/main/java/com/strv/movies/data/database/MoviesDataStore.kt [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    val darkModeFlow: Flow<Boolean> = context.dataStore.data
+        .map { preferences ->
+            preferences[DARK_THEME_ENABLED] ?: false
+        }
+
+    suspend fun setDarkTheme(isDarkTheme: Boolean) {
+        context.dataStore.edit { preferences ->
+            preferences[DARK_THEME_ENABLED] = isDarkTheme
+        }
+    }
```
Comment (jipariz) [inline app/src/main/java/com/strv/movies/data/database/MoviesDataStore.kt:35]: I would like more it like this, but that's just my preference :)
```suggestion
    private val dataStore = PreferenceDataStoreFactory.create() {
        context.preferencesDataStoreFile(MOVIES_APP_DATASTORE)
    }

    val darkModeFlow: Flow<Boolean> = dataStore.data
        .map { preferences ->
            preferences[DARK_THEME_ENABLED] ?: false
        }

    suspend fun setDarkTheme(isDarkTheme: Boolean) {
        dataStore.edit { preferences ->
            preferences[DARK_THEME_ENABLED] = isDarkTheme
        }
    }
```

**Review by jipariz — COMMENTED** [review]
Nice 👍 I have just two comments

File: app/src/main/java/com/strv/movies/MainViewModel.kt (hunk 1 above)
Comment (JakubGongol) [reply app/src/main/java/com/strv/movies/MainViewModel.kt]: 👍

File: app/src/main/java/com/strv/movies/data/database/MoviesDataStore.kt (hunk 2 above)
Comment (JakubGongol) [reply app/src/main/java/com/strv/movies/data/database/MoviesDataStore.kt]: 👍

**Review by jipariz — APPROVED** [review]
Good to go 👍

**Pull request merged**